### ISSUE TITLE
Invalid Filter clause error when requesting NextPage of a filtered result

### ISSUE THREAD

**Issue by MichaMican** [microsoftgraph/msgraph-sdk-dotnet#1095]
**Describe the bug**
When doing a filtered request with the SDK against the GraphAPI, which is paged the `result.NextPageRequest.GetAsync()` returns an error:
_Code: BadRequest
Message: Invalid filter clause_

**To Reproduce**
1. Get a GraphServiceClient object (in my case I used the Microsoft.Graph.Auth library to get a ClientCredentialProvider)
2. Request Users with `var result = await graphClient.Users.Request().Filter("startswith(displayName, 'a') or startswith(UserPrincipalName, a')").GetAsync();` (note that you'll need to request this against a tenant where more than 100 Users have a displayname or a UserPrincipalName that starts with 'a')
3. check for `result.NextPageRequest != null` and call `result = await result.NextPageRequest.GetAsync()`

**Expected behavior**
It would be expected, that I get the next page of results for users starting with 'a'

**Desktop (please complete the following information):**
 - OS: Windows 10 and Azure AppService
 - Browser: any [issue is backend related]
 - Version: MSGraph 4.1.0

**Additional context**
This issue wasn't present in MSGraph SDK version 3.0.0
Same issue when using PageIterator

**Comment by andrueastman** [comment]
Thanks for raising this @MichaMican. 

For now you can work around this by 
```cs
  var result = await graphServiceClient.Users.Request().Filter("startswith(displayName, 'a') or startswith(UserPrincipalName, a')").GetAsync();
  var newFilter = result.NextPageRequest?.QueryOptions.FirstOrDefault(x => x.Name.EndsWith("filter"));
  if (newFilter != null)
  {
      var index = result.NextPageRequest.QueryOptions.IndexOf(newFilter);
      result.NextPageRequest.QueryOptions[index] = new QueryOption(newFilter.Name, newFilter.Value.Replace("+", " "));
  }

  var nextPage = await result.NextPageRequest.GetAsync();
```

This should be resolved/fixed in 4.2.0 which will be available by the end of the week.

**Comment by ThreeSevenths** [comment]
@andrueastman is this fixed in the Beta 4.7.0 package? If not, will a fix be incoming for beta as well?

**Comment by andrueastman** [comment]
The issue is also in the beta package but should be fixed the next beta release that will be out together with the v1.0 release.

**Comment by andrueastman** [comment]
Closing this as this is now resolved via the [4.2.0 release](https://www.nuget.org/packages/Microsoft.Graph/4.2.0) (and the [4.8.0-preview](https://www.nuget.org/packages/Microsoft.Graph.Beta/4.8.0-preview) release for beta).
